Penteract/Assets/Scripts/SceneTransition.cpp

```diff
@@ -0,0 +1,103 @@
+#include "SceneTransition.h"
+
+#include "Components/UI/ComponentTransform2D.h"
+#include "GameObject.h"
+#include "GameplaySystems.h"
+
+#define FADE_SPEED 0.01f
+#define CENTER_POSITION 0.f
+#define ALPHA_TRANSPARENCY 255.f
+#define FULL_OPACY 1.f
+#define FULL_TRANSPARENCY 0.f
+
+EXPOSE_MEMBERS(SceneTransition) {
+	MEMBER(MemberType::SCENE_RESOURCE_UID, sceneUID),
+	MEMBER(MemberType::GAME_OBJECT_UID, transitionUID),
+	MEMBER(MemberType::INT, transitionMove),
+	MEMBER(MemberType::FLOAT, speedTransition),
+};
+
+GENERATE_BODY_IMPL(SceneTransition);
+
+void SceneTransition::Start() {
+	actualResolution = Screen::GetResolution();
+	transitionGO = GameplaySystems::GetGameObject(transitionUID);
+	if (transitionGO) {
+		transform2D = transitionGO->GetComponent<ComponentTransform2D>();
+		image2D = transitionGO->GetComponent<ComponentImage>();
+	}
+}
+
+void SceneTransition::Update() {
+	if (startTransition) {
+		if (finishedTransition) {
+			if (sceneUID != 0) {
+				isExit ? SceneManager::ExitGame() : SceneManager::ChangeScene(sceneUID);
+				if (Time::GetDeltaTime() == 0.f) Time::ResumeGame();
+			}
+		}
+		else {
+			// Transition
+			// Size should be equal then size of the screen
+			if (transitionMove == (int)TransitionMove::LEFT_TO_RIGHT) {
+				transform2D->SetPosition(float3(transform2D->GetPosition().x + (1.f * speedTransition), 0.f, 0.f));
```

Comment:
If the GameObject with the transition hasn't had a ComponentImage this line will crash. Add a check of nullity at the beginning of the update.

Reply:
Fixed checking if the GameObject has a ComponentTransform2D and ComponentImage at the beginning of the update ;)